compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -378,11 +378,13 @@ impl<'f> PerFunctionContext<'f> {
 
                 self.check_array_aliasing(references, value);
 
+                // This causes errors in the sha256 tests
+                //
                 // If there was another store to this instruction without any (unremoved) loads or
                 // function calls in-between, we can remove the previous store.
-                if let Some(last_store) = references.last_stores.get(&address) {
-                    self.instructions_to_remove.insert(*last_store);
-                }
+                // if let Some(last_store) = references.last_stores.get(&address) {
+                //     self.instructions_to_remove.insert(*last_store);
+                // }
```

Comment:
Had to comment out this portion of mem2reg as it was causing the sha256 tests to fail. There seems to be an issue with tracking `last_stores` for some reason now.